ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -83,10 +80,11 @@ internal constructor(
   surveyRepository: SurveyRepository,
 ) : AbstractViewModel() {
 
-  private val loiId: String? = savedStateHandle["locationOfInterestId"]
+  private val jobId: String? = savedStateHandle[TASK_JOB_ID_KEY]
+  private val loiId: String? = savedStateHandle[TASK_LOI_ID_KEY]
+
   private val activeSurvey: Survey = requireNotNull(surveyRepository.activeSurvey)
-  private val job: Job =
-    activeSurvey.getJob(requireNotNull(savedStateHandle["jobId"])) ?: error("empty job")
+  private val job: Job = activeSurvey.getJob(requireNotNull(jobId)) ?: error("empty job")
```

Comment:
Instead of just "empty job" here let's indicate we couldn't retrieve the job for `$jobId`

Reply:
Done